File: src/Deviser.Core/Deviser.Admin.Web/ClientApp/src/app/common/domain-types/date-filter.ts
import { Filter } from './filter';
import { DateTimeOperator } from './date-time-operator';

export class DateFilter implements Filter {
    fieldName: string;
    operator: DateTimeOperator;
    date: Date;
    fromDate: Date;
    toDate: Date;
    constructor(fieldName: string) {
        this.fieldName = fieldName;
    }
}

File: src/Deviser.Core/Deviser.Admin/Config/Filters/DateFilter.cs
using System;
using System.Collections.Generic;
using System.Text;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

namespace Deviser.Admin.Config.Filters
{
    public class DateFilter : Filter
    {
        [JsonConverter(typeof(StringEnumConverter))]
        public DateTimeOperator Operator { get; set; }
        public DateTime? Date { get; set; }
        public DateTime? FromDate { get; set; }
        public DateTime? ToDate { get; set; }
    }
}